File: apps/client/app/features/admin/types/AdminUsage.ts
import type { BudgetUsageStatus } from '../../budget/types';

export type AdminWorkflowBudgetUsage = {
  monthly_budget_usd: number;
  current_month_cost: number;
  usage_ratio: number;
  status: BudgetUsageStatus | string;
  usage_data_complete: boolean;
  unresolved_provider_call_count: number;
};

export type AdminWorkflowUsageItem = {
  workflow_id: string;
  workflow_name: string;
  prompt_tokens: number;
  completion_tokens: number;
  call_count: number;
  total_cost: number;
  workflow_execution_cost: number;
  agent_builder_cost: number;
  usage_data_complete: boolean;
  unresolved_provider_call_count: number;
  budget?: AdminWorkflowBudgetUsage | null;
};

export type AdminUsagePeriod = {
  startAt: string;
  endAt: string;
};

export type AdminWorkflowUsageResponse = {
  total: number;
  period: AdminUsagePeriod;
  usage_data_complete: boolean;
  unresolved_provider_call_count: number;
  items: AdminWorkflowUsageItem[];
};

export type AdminBudgetSummary = {
  budgeted_workflow_count: number;
  at_risk_count: number;
  exceeded_count: number;
  ratio: number;
};

export type AdminOrganizationSummary = {
  month: string;
  total_cost: number;
  workflow_execution_cost: number;
  agent_builder_cost: number;
  usage_data_complete: boolean;
  unresolved_provider_call_count: number;
  budget: AdminBudgetSummary | null;
};

File: apps/shared/schemas/admin_usage.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class AdminUsagePeriodResponse(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    start_at: datetime = Field(alias="startAt")
    end_at: datetime = Field(alias="endAt")


class AdminWorkflowBudgetBlock(BaseModel):
    monthly_budget_usd: float
    current_month_cost: float
    usage_ratio: float
    status: str
    usage_data_complete: bool = True
    unresolved_provider_call_count: int = 0


class AdminBudgetSummaryBlock(BaseModel):
    budgeted_workflow_count: int
    at_risk_count: int
    exceeded_count: int
    ratio: float


class AdminWorkflowUsageItem(BaseModel):
    workflow_id: UUID
    workflow_name: str
    prompt_tokens: int
    completion_tokens: int
    call_count: int
    total_cost: float
    workflow_execution_cost: float
    agent_builder_cost: float
    usage_data_complete: bool = True
    unresolved_provider_call_count: int = 0
    budget: AdminWorkflowBudgetBlock | None = None


class AdminWorkflowUsageResponse(BaseModel):
    total: int
    period: AdminUsagePeriodResponse
    usage_data_complete: bool = True
    unresolved_provider_call_count: int = 0
    items: list[AdminWorkflowUsageItem]


class AdminOrganizationSummaryResponse(BaseModel):
    month: str
    total_cost: float
    workflow_execution_cost: float
    agent_builder_cost: float
    usage_data_complete: bool = True
    unresolved_provider_call_count: int = 0
    budget: AdminBudgetSummaryBlock | None = None
